subsys/mgmt/mcumgr/grp/img_mgmt/src/img_mgmt.c

```diff
@@ -141,9 +141,10 @@ static int img_mgmt_find_tlvs(int slot, size_t *start_off, size_t *end_off, uint
 
 int img_mgmt_active_slot(int image)
 {
-#if CONFIG_MCUMGR_GRP_IMG_UPDATABLE_IMAGE_NUMBER == 2
-	if (image == 1) {
-		return 2;
+	LOG_DBG("(%d)", image);
+#if CONFIG_MCUMGR_GRP_IMG_UPDATABLE_IMAGE_NUMBER > 1
+	if (image >= 1) {
+		return (image << 1);
```

Comment:
Not requesting a change but this will probably cause a problem in the future because it assumes that the primary slot in other images is the one executing, but if other images are using direct XIP then this is an incorrect assumption

Reply:
Yes, that is known problem. Without communicating with other app or pulling info from MCUboot, for example via retention, it is hard to tell what is actually active slot of another image as what is set to flash may not yet be actually executing.